src/components/App.tsx

```diff
@@ -38,26 +65,33 @@ function App() {
 
   const onSubmit: SubmitHandler<Input> = (data) => {
     if (currentAction.input.includes(data.userInput)) {
-      setResult(Result.WIN);
+      if (points !== 0 && points % 4 === 0) {
```

Comment:
Here you should be ensure that `level` isn't exceeding the max level available. If `level` goes higher than 6, the app crash.